### PR TITLE
[Modular] NRI MODsuit

### PR DESCRIPTION
**Pull request by Stalkeros2**
## About The Pull Request
Adds NRI MODsuit for ERT and NRI Raiders, as well as a new MODsuit module that heals wounds if your health is lower than 80 and it has enough chems inside. Medical module is limited to NRI ERT only, NRI Raiders do not get it as they have a completely different set of modules.
Makes VOSKHODs **obsolete** and removes them from everywhere but the code.
cool dimly lit screenshots because I am a moron
image
image
image
image
image
No medical module screenshot because it literally uses ninja adrenaline module's sprite.

## How This Contributes To The Skyrat Roleplay Experience
MODs are a way better option to VOSKHODs (that were a literal placeholder when they were used by NRI Raiders) and offer way more flexibility without being terribly overpowered. Also, this is a nerf to NRI Raiders' combat capabilities because they're no longer basically immortal.

## Changelog

:cl: Stalkeros, Lukaster42
add: NRI MODsuit in two variations: one for ERT, other for raiders.
add: MOD automatic paramedical module.
add: generic non-Ninja-related MOD status readout module.
del: Pirate VOSKHODs; technically, VOSKHODs in their (available) entirety, since ERT now uses MODs.
/:cl:

### CODE REVIEW

**Comment by Stalkeros2** [comment]
> You currently have a negative Fix/Feature pull request delta of -17. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

neco-arc

File: modular_skyrat/modules/novaya_ert/code/mod_suit.dm [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+/obj/item/mod/control/pre_equipped/frontline
+	theme = /datum/mod_theme/frontline
+	initial_modules = list(
+		/obj/item/mod/module/storage,
+		/obj/item/mod/module/flashlight)
+
+/obj/item/mod/control/pre_equipped/frontline/pirate
+	initial_modules = list(
+		/obj/item/mod/module/storage/large_capacity,
+		/obj/item/mod/module/thermal_regulator,
```
Comment (tf-4) [inline modular_skyrat/modules/novaya_ert/code/mod_suit.dm:66]: Aren't they human only?

File: modular_skyrat/modules/novaya_ert/code/mod_suit.dm [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+		/obj/item/mod/module/status_readout/generic,
+		/obj/item/mod/module/visor/night,
+		/obj/item/mod/module/emp_shield,
+		/obj/item/mod/module/jetpack,
+		/obj/item/mod/module/tether,
```
Comment (tf-4) [inline modular_skyrat/modules/novaya_ert/code/mod_suit.dm]: I'd do one or the other, both is odd

Comment (Zonespace27) [reply modular_skyrat/modules/novaya_ert/code/mod_suit.dm]: iirc they are meant to be incompatible

File: modular_skyrat/modules/novaya_ert/code/mod_suit.dm (hunk 1 above)
Comment (Stalkeros2) [reply modular_skyrat/modules/novaya_ert/code/mod_suit.dm:66]: No digi sprites rn, but no, they are not.

**Comment by Stalkeros2** [comment]
I'll need to throw in screenshots and descriptions for both the MOD and PR tomorrow, but other than that, it's completely ready code-wise. Because making a proper directional shield module will be painful.

**Comment by Stalkeros2** [comment]
It's now completely ready codewise.
WISH I COULD WRITE LONG TEXTS FOR EXTENDED DESCS THO :((((((((((((((((
Will probably do later.

File: modular_skyrat/modules/novaya_ert/code/mod_suit.dm (hunk 2 above)
Comment (Stalkeros2) [reply modular_skyrat/modules/novaya_ert/code/mod_suit.dm]: Replaced with tether only. Iirc they're not incompatible but anyways.

File: modular_skyrat/modules/novaya_ert/code/mod_suit.dm [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+		/obj/item/flashlight,
+		/obj/item/tank/internals,
+		/obj/item/ammo_box,
+		/obj/item/ammo_casing,
+		/obj/item/restraints/handcuffs,
+		/obj/item/assembly/flash,
+		/obj/item/melee/baton,
+		/obj/item/knife/combat,
+		/obj/item/shield/riot,
+		/obj/item/gun
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/novaya_ert/code/mod_suit.dm]:
```suggestion
		/obj/item/gun,
```

File: modular_skyrat/modules/novaya_ert/code/mod_suit.dm [hunk 4]
```diff
@@ -0,0 +1,223 @@
+/datum/mod_theme/frontline
+	name = "frontline"
+	desc = "A Novaya Rossiyskaya Imperiya Defense Collegia protective suit, designed for fortified positions operation and humanitarian aid."
+	extended_desc = "PLACEHOLDER WOOOOOOOOOOOOOOOOOOOOOOO"
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/novaya_ert/code/mod_suit.dm]: If you have nothing to put there, don't put anything.

File: modular_skyrat/modules/novaya_ert/code/mod_suit.dm [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+				CAN_OVERSLOT = TRUE,
+			),
+		),
+	)
+
+/obj/item/mod/control/pre_equipped/frontline
+	theme = /datum/mod_theme/frontline
+	initial_modules = list(
+		/obj/item/mod/module/storage,
+		/obj/item/mod/module/flashlight)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/novaya_ert/code/mod_suit.dm]:
```suggestion
		/obj/item/mod/module/flashlight,
	)
```

File: modular_skyrat/modules/novaya_ert/code/mod_suit.dm [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+
+/obj/item/mod/control/pre_equipped/frontline/pirate
+	initial_modules = list(
+		/obj/item/mod/module/storage/large_capacity,
+		/obj/item/mod/module/thermal_regulator,
+		/obj/item/mod/module/status_readout/generic,
+		/obj/item/mod/module/tether,
+		/obj/item/mod/module/magboot
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/novaya_ert/code/mod_suit.dm]:
```suggestion
		/obj/item/mod/module/magboot,
```

File: modular_skyrat/modules/novaya_ert/code/mod_suit.dm [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+/obj/item/mod/control/pre_equipped/frontline/ert
+	applied_cell = /obj/item/stock_parts/cell/hyper
+	initial_modules = list(
+		/obj/item/mod/module/storage/syndicate,
+		/obj/item/mod/module/thermal_regulator,
+		/obj/item/mod/module/status_readout/generic,
+		/obj/item/mod/module/auto_doc,
+		/obj/item/mod/module/visor/thermal,
+		/obj/item/mod/module/jetpack/advanced,
+		/obj/item/mod/module/magboot/advanced
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/novaya_ert/code/mod_suit.dm]:
```suggestion
		/obj/item/mod/module/magboot/advanced,
```

File: modular_skyrat/modules/novaya_ert/code/mod_suit.dm [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+		Using a built-in storage of chemical compounds and miniature surgical robots capable of pinpoint damage treatment,\
+		it's capable of injecting its user with simple painkillers and coagulants, assisting them with their restoration, as long as they don't overdose themselves. \
+		However, this system heavily relies on some rarely combat-available chemical compounds to prepare its injections, mainly Cryptobiolin, which appear in the user's bloodstream from time to time, \
+		and its trivial damage assesment systems are inadequate for complete restoration purposes."
+	icon_state = "adrenaline_boost"
+	module_type = MODULE_TOGGLE
+	incompatible_modules = list(/obj/item/mod/module/adrenaline_boost, /obj/item/mod/module/auto_doc)
+	cooldown_time = null
+	complexity = 4
+	use_power_cost = DEFAULT_CHARGE_DRAIN*20
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/novaya_ert/code/mod_suit.dm]:
```suggestion
	use_power_cost = DEFAULT_CHARGE_DRAIN * 20
```

File: modular_skyrat/modules/novaya_ert/code/mod_suit.dm [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+	if(new_stamloss)
+		mod.wearer.adjustStaminaLoss(-stamina_heal_amount)
+		mod.wearer.reagents.add_reagent(/datum/reagent/medicine/mine_salve, 10)
+		mod.wearer.reagents.add_reagent(/datum/reagent/medicine/stimulants, 10)
+		to_chat(mod.wearer, span_warning("Combat stimulants administered. Overdose risks present on further use, consult your first-aid analyzer."))
+	mod.wearer.reagents.add_reagent(/datum/reagent/medicine/coagulant, 5)
+	reagents.remove_reagent(reagent_required, reagent_required_amount)
+	drain_power(use_power_cost*10)
+	addtimer(CALLBACK(src, .proc/boost_aftereffects, mod.wearer), 45 SECONDS)
+	COOLDOWN_START(src, heal_timer, heal_cooldown)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/novaya_ert/code/mod_suit.dm]: This is so hard to read, let it breathe.

I also feel like the stuff that keeps on getting repeated in those last few ifs should be made into a proc instead.

Also, why is it both _healing_ *and* _injecting reagents_? That feels a tad overpowered, and by a tad, I mean very very strong.
```suggestion
	if(!COOLDOWN_FINISHED(src, heal_timer))
		return FALSE

	if(!check_power(use_power_cost))
		balloon_alert(mod.wearer, "not enough charge!")
		SEND_SIGNAL(src, COMSIG_MODULE_DEACTIVATED)
		return FALSE

	if(!allowed_in_phaseout && istype(mod.wearer.loc, /obj/effect/dummy/phased_mob))
		to_chat(mod.wearer, span_warning("You cannot activate this right now."))
		return FALSE

	if(SEND_SIGNAL(src, COMSIG_MODULE_TRIGGERED) & MOD_ABORT_USE)
		return FALSE

	if(!reagents.has_reagent(reagent_required, reagent_required_amount))
		balloon_alert(mod.wearer, "not enough chems!")
		SEND_SIGNAL(src, COMSIG_MODULE_DEACTIVATED)
		return FALSE

	if(mod.wearer.health > health_threshold)
		return

	var/new_bruteloss = mod.wearer.getBruteLoss()
	var/new_fireloss = mod.wearer.getFireLoss()
	var/new_toxloss = mod.wearer.getToxLoss()
	var/new_stamloss = mod.wearer.getStaminaLoss()
	playsound(mod.wearer, 'modular_skyrat/modules/hev_suit/sound/hev/hiss.ogg', 100)

	if(new_bruteloss)
		mod.wearer.adjustBruteLoss(-heal_amount)
		mod.wearer.reagents.add_reagent(/datum/reagent/medicine/mine_salve, 10)
		mod.wearer.reagents.add_reagent(/datum/reagent/medicine/sal_acid, 5)
		to_chat(mod.wearer, span_warning("Brute treatment administered. Overdose risks present on further use, consult your first-aid analyzer."))

	if(new_fireloss)
		mod.wearer.adjustFireLoss(-heal_amount)
		mod.wearer.reagents.add_reagent(/datum/reagent/medicine/mine_salve, 10)
		mod.wearer.reagents.add_reagent(/datum/reagent/medicine/oxandrolone, 5)
		to_chat(mod.wearer, span_warning("Burn treatment administered. Overdose risks present on further use, consult your first-aid analyzer."))

	if(new_toxloss)
		mod.wearer.adjustToxLoss(-heal_amount)
		mod.wearer.reagents.add_reagent(/datum/reagent/medicine/mine_salve, 10)
		mod.wearer.reagents.add_reagent(/datum/reagent/medicine/pen_acid, 5)
		to_chat(mod.wearer, span_warning("Toxin treatment administered. Overdose risks present on further use, consult your first-aid analyzer."))

	if(new_stamloss)
		mod.wearer.adjustStaminaLoss(-stamina_heal_amount)
		mod.wearer.reagents.add_reagent(/datum/reagent/medicine/mine_salve, 10)
		mod.wearer.reagents.add_reagent(/datum/reagent/medicine/stimulants, 10)
		to_chat(mod.wearer, span_warning("Combat stimulants administered. Overdose risks present on further use, consult your first-aid analyzer."))

	mod.wearer.reagents.add_reagent(/datum/reagent/medicine/coagulant, 5)
	reagents.remove_reagent(reagent_required, reagent_required_amount)
	drain_power(use_power_cost*10)
	addtimer(CALLBACK(src, .proc/boost_aftereffects, mod.wearer), 45 SECONDS)
	COOLDOWN_START(src, heal_timer, heal_cooldown)
```

File: modular_skyrat/modules/novaya_ert/code/mod_suit.dm [hunk 10]
```diff
@@ -0,0 +1,223 @@
...
+/obj/item/mod/module/auto_doc/proc/charge_boost(obj/item/attacking_item, mob/user)
+	if(!attacking_item.is_open_container())
+		return FALSE
+	if(reagents.has_reagent(reagent_required, reagent_max_amount))
+		balloon_alert(mod.wearer, "already full!")
+		return FALSE
+	if(!attacking_item.reagents.trans_id_to(src, reagent_required, reagent_required_amount))
+		return FALSE
+	balloon_alert(mod.wearer, "charge reloaded")
+	return TRUE
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/novaya_ert/code/mod_suit.dm]: Let it breathe, also, please add documentation here.
```suggestion
/obj/item/mod/module/auto_doc/proc/charge_boost(obj/item/attacking_item, mob/user)
	if(!attacking_item.is_open_container())
		return FALSE

	if(reagents.has_reagent(reagent_required, reagent_max_amount))
		balloon_alert(mod.wearer, "already full!")
		return FALSE

	if(!attacking_item.reagents.trans_id_to(src, reagent_required, reagent_required_amount))
		return FALSE

	balloon_alert(mod.wearer, "charge reloaded")
	return TRUE
```

File: modular_skyrat/modules/novaya_ert/code/mod_suit.dm [hunk 11]
```diff
@@ -0,0 +1,223 @@
...
+	if(charge_boost(attacking_item, user))
+		return COMPONENT_NO_AFTERATTACK
+	return NONE
+
+/obj/item/mod/module/auto_doc/proc/boost_aftereffects(mob/affected_mob)
+	if(!affected_mob)
+		return
+	mod.wearer.reagents.add_reagent(/datum/reagent/cryptobiolin, 10)
+	mod.wearer.reagents.add_reagent(/datum/reagent/drug/maint/sludge, 5)
+	to_chat(affected_mob, span_danger("Your head starts slightly spinning, and your chest hurts."))
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/novaya_ert/code/mod_suit.dm]: Also add some documentation here please.
```suggestion
/obj/item/mod/module/auto_doc/proc/on_attackby(datum/source, obj/item/attacking_item, mob/user)
	SIGNAL_HANDLER

	if(charge_boost(attacking_item, user))
		return COMPONENT_NO_AFTERATTACK

	return NONE

/obj/item/mod/module/auto_doc/proc/boost_aftereffects(mob/affected_mob)
	if(!affected_mob)
		return

	mod.wearer.reagents.add_reagent(/datum/reagent/cryptobiolin, 10)
	mod.wearer.reagents.add_reagent(/datum/reagent/drug/maint/sludge, 5)
	to_chat(affected_mob, span_danger("Your head starts slightly spinning, and your chest hurts."))
```

File: modular_skyrat/modules/novaya_ert/code/outfit.dm [hunk 12]
```diff
@@ -1,19 +1,21 @@
...
-	suit = /obj/item/clothing/suit/space/hev_suit/nri
-	suit_store = /obj/item/gun/ballistic/automatic/akm
+	suit = null
+	suit_store = null
 	gloves = /obj/item/clothing/gloves/combat
 	belt = /obj/item/storage/belt/military/nri/soldier
-	back = /obj/item/storage/backpack/duffelbag/syndie/nri
-	backpack_contents = list(/obj/item/storage/box/nri_survival_pack, /obj/item/storage/medkit/emergency, /obj/item/ammo_box/magazine/m9mm_aps, /obj/item/advanced_choice_beacon/nri/heavy, /obj/item/beamout_tool, /obj/item/crucifix)
+	back = /obj/item/mod/control/pre_equipped/frontline/ert
+	backpack_contents = list(/obj/item/storage/box/nri_survival_pack, /obj/item/storage/medkit/emergency, /obj/item/ammo_box/magazine/m9mm_aps, /obj/item/advanced_choice_beacon/nri/heavy, /obj/item/beamout_tool, /obj/item/crucifix, /obj/item/reagent_containers/cup/glass/waterbottle/large/cryptobiolin)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/novaya_ert/code/outfit.dm]: Make these multi-line lists please, and end the last element with a trailing comma, to lessen the impact on the git blame of any additions made to those lists in the future.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
Also, real talk, what even is cryptobiolin, and why is it everywhere in this PR?

**Comment by Stalkeros2** [comment]
@GoldenAlpharex, Cryptobiolin is a chemical used in the spaceacillin's recipe and that doesn't do much on its own. Here it is used to refill the medical module. Giving it a stronger chemical requirement doesn't make much sense because this is an ERT module, and so it can possibly use some more available chems. 
I'll check the review in a while.

File: modular_skyrat/modules/novaya_ert/code/mod_suit.dm (hunk 4 above)
Comment (Stalkeros2) [reply modular_skyrat/modules/novaya_ert/code/mod_suit.dm]: I was totally expecting getting sent the replacement description by a person that I've asked but oh well. I'll remove it whenever possible.

File: modular_skyrat/modules/novaya_ert/code/mod_suit.dm (hunk 9 above)
Comment (Stalkeros2) [reply modular_skyrat/modules/novaya_ert/code/mod_suit.dm]: I've designed it with only the ERT in mind, and thus didn't care *that* much about the balancing. I'll remove the damage healing part; leaving chemicals.

File: modular_skyrat/modules/novaya_ert/code/mod_suit.dm (hunk 4 above)
Comment (Stalkeros2) [reply modular_skyrat/modules/novaya_ert/code/mod_suit.dm]: Done.

**Comment by Stalkeros2** [comment]
I *think* that's it.

File: modular_skyrat/modules/encounters/code/enclave_pirates.dm [hunk 13]
```diff
@@ -205,13 +187,10 @@
 	<br> <span style=\"color:black;font-family:'Segoe Script';\"><p><b>Voronov Platon Aleksandrovich and Radich Katarina Dinovich.</b></p></span>"}
 
 /obj/machinery/suit_storage_unit/nri
-	suit_type = /obj/item/clothing/suit/space/hev_suit/nri/pirate
-	helmet_type = /obj/item/clothing/head/helmet/space/hev_suit/nri
+	mod_type = /obj/item/mod/control/pre_equipped/frontline/pirate
 	storage_type = /obj/item/tank/internals/oxygen/yellow
 
 /obj/machinery/suit_storage_unit/nri/captain
```
Comment (Zonespace27) [inline modular_skyrat/modules/encounters/code/enclave_pirates.dm:193]: what is the point of this

File: modular_skyrat/modules/encounters/code/enclave_pirates.dm [hunk 14]
```diff
@@ -23,7 +23,7 @@
 
 	belt = /obj/item/storage/belt/military/nri/captain/pirate_officer
 	back = /obj/item/storage/backpack/duffelbag/syndie/nri/captain
-	backpack_contents = list(/obj/item/storage/box/nri_survival_pack = 1, /obj/item/ammo_box/magazine/automag = 3, /obj/item/gun/ballistic/automatic/pistol/automag = 1, /obj/item/crucifix = 1, /obj/item/device/traitor_announcer = 1, /obj/item/clothing/mask/gas/hecu2 = 1, /obj/item/modular_computer/tablet/pda/security = 1)
+	backpack_contents = list(/obj/item/storage/box/nri_survival_pack = 1, /obj/item/ammo_box/magazine/automag = 3, /obj/item/gun/ballistic/automatic/pistol/automag = 1, /obj/item/crucifix = 1, /obj/item/device/traitor_announcer = 1, /obj/item/clothing/mask/gas/hecu2 = 1, /obj/item/modular_computer/tablet/pda/security = 1, /obj/item/reagent_containers/cup/glass/waterbottle/large/cryptobiolin = 1)
```
Comment (Zonespace27) [inline modular_skyrat/modules/encounters/code/enclave_pirates.dm]: cryptobiolin is pointless as the NRI do not get autodocs

File: modular_skyrat/modules/encounters/code/enclave_pirates.dm (hunk 13 above)
Comment (Stalkeros2) [reply modular_skyrat/modules/encounters/code/enclave_pirates.dm:193]: I'm too lazy to make map edits to remove it in this PR; will be gone when I nerf the raider ship's cintents.

File: modular_skyrat/modules/encounters/code/enclave_pirates.dm (hunk 14 above)
Comment (Stalkeros2) [reply modular_skyrat/modules/encounters/code/enclave_pirates.dm]: Oversight, as their SL initially was meant to have the ERT suit, but then I thought that was a bad idea and scrapped that. I'll remove it soon enough.

**Pull request merged**